pkg/appender/appender.go

```diff
@@ -217,6 +218,10 @@ func (mc *MetricsCache) Add(lset utils.LabelsIfc, t int64, v interface{}) (uint6
 
 	if !ok {
 		metric = &MetricState{Lset: lset, key: key, name: name, hash: hash}
+		// if the (first) value is not float, use variant encoding, TODO: test w schema
+		if _, ok := v.(float64); !ok {
```

Comment:
Maybe you should take care of `NaN`. With `NaN` you can't determine the type, so it's better to skip it

Reply:
Discussed in person